src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/OverhangFixingManager.java

```diff
@@ -307,10 +385,49 @@ public void setRead(final GATKRead read) {
                 }
             }
         }
+
+        // Returns true if either of the required mate information fields have been changed by the clipper
+        public boolean hasBeenOverhangClipped() {
+            return (!oldCigar.equals(read.getCigar())) || (oldStart != read.getStart());
+        }
+
+        // Adds the relevant information for repairing the mate to setMateChanged keyed on a string composed of the start position
+        public void setMateChanged() {
+            if (!read.isUnmapped()) {
+                mateChangedReads.put(read.getName() + (read.isFirstOfPair()? 0: 1) + oldStart,
+                        new Tuple<>(read.getStart(),read.getCigar().toString()));
```

Comment:
use `TextCigarCodec.encode()` instead of toString.  Better to not rely on the behavior of toString for program correctness.